protocols/v2/roles-logic-sv2/src/selectors.rs

```diff
@@ -65,14 +65,56 @@ impl<Down: IsMiningDownstream> DownstreamMiningSelector<Down>
     }
 
     fn remove_downstreams_in_channel(&mut self, channel_id: u32) -> Vec<Arc<Mutex<Down>>> {
```

Comment:
It looks like `remove_downstream` and `_remove_downstream` have been added to this trait solely for the internal implementation of this function - basing that assessment on the other implementation of this trait invoking `unreachable!` in the implementation, and these functions not being invoked anywhere else in this pull request.

If that's the case, perhaps the functions might have appropriately restricted visibility by either declaring them inside of `remove_downstreams_in_channel` (ie. nested functions), or declaring them in a `impl<Down: IsMiningDownstream> ProxyDownstreamMiningSelector<Down> {` block outside the trait implementation?

Let me know if I missed something...

Reply:
You are right no need to have it in the trait, I put it there trying to make things works and then left there.